internal/cmd/eventgen/templates/linkslice_decl.tmpl

```diff
@@ -11,3 +13,26 @@ func (links *{{.TypeName}}) Add(linkType string, target MetaTeller) {
 func (links *{{.TypeName}}) AddByID(linkType string, target string) {
 	*links = append(*links, {{.SlicedType}}{Target: target, Type: linkType})
 }
+
+// FindAll returns the IDs of all links of the specified type, or an empty
+// slice if no such links are found.
+func (links {{.TypeName}}) FindAll(linkType string) []string {
+	result := make([]string, 0, len(links))
+	for _, link := range links {
+		if link.Type == linkType {
+			result = append(result, link.Target)
+		}
+	}
+	return result
+}
+
+// FindFirst returns the ID of the first encountered link of the specified
+// type, or an empty string if no such link is found.
+func (links {{.TypeName}}) FindFirst(linkType string) string {
```

Comment:
Nit; I am not really sure I see the benefit of this one over just FindAll("CONTEXT")[0]. I might be missing something, but it feels like I would probably want to do a len() check of the links and handle the case if there are more than one. 
In both cases we also need a 0 check in case nothing is returned.

Reply:
That's a reasonable point. You're right that we could totally make do with just FindAll and the benefit of FindFirst is limited. In the places in existing code where I called FindFirst I used the result as a map key, allowing me to do

```go
val, found := somemap[event.Links.FindFirst("CAUSE")]
if !found {
        ...
}
```

instead of

```go
ids := event.Links.FindFirst("CAUSE")
if len(ids) == 0 {
        ...
}
val, found := somemap[ids[0]]
if !found {
        ...
}
```

so it's not always just a matter of having to tack on `[0]` to the end of the return value or call len() instead of just checking for an empy string.

I think the extra method is justifiable since FindFirst in my experience will be the most commonly used LinkFinder method, but as I note in the Possible Drawbacks section there's a risk that too much convenience fluff will be added over time, creating an enormous API in the end. As long as convenience helpers are kept small and reusable I think we can manage that risk.